framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/OpenShiftDeployment.java

```diff
@@ -88,38 +101,85 @@ public boolean isReady() {
     @Override
     public List<Instance> getInstances() {
         if (isReady()) {
-            String deploymentConfigName = openShiftController.getProject(namespace).getService(getServiceName()).getDeploymentConfig().getName();
-            List<Pod> pods = openShiftController.getClient().pods().inNamespace(namespace).withLabel(OpenShiftResourceConstants.DEPLOYMENT_CONFIG_LABEL, deploymentConfigName).list().getItems();
+            // Deployment config has a same name as its service.
+            String deploymentConfigName = getServiceName();
 
-            List<Instance> instances = pods.stream().map((pod) -> {
-                return createInstance(pod);
-            }).collect(toList());
+            List<Instance> instances = util.getPods().stream()
+                    .filter(pod -> pod.getMetadata().getLabels().get(OpenShiftResourceConstants.DEPLOYMENT_CONFIG_LABEL) != null
+                                && pod.getMetadata().getLabels().get(OpenShiftResourceConstants.DEPLOYMENT_CONFIG_LABEL).equals(deploymentConfigName))
```

Comment:
I'd rather use
```
 deploymentConfigName.equals(
  pod.getMetadata().getLabels().get(OpenShiftResourceConstants.DEPLOYMENT_CONFIG_LABEL)
 )
```

to avoid having to call the thing twice. 
And (optional) I'd rename the variable to `deploymentConfigLabel` to make it clear we're comparing apples to apples and not to pears :)

Reply:
It is reworked for better readability.
I kept there deploymentConfigName as IMHO it better describes what does we compare there. In filter predicate we check that name of deployment config (stored in pod's label) is same as actual deployment config name - to find out what pod belongs to the deployment config.